variants.go

```diff
@@ -5,6 +5,30 @@ import "time"
 // Variants is a collection of variants
 type Variants []Variant
 
+// GetByPosition returns a variant at the found position or throws an error if not found
+func (v Variants) GetByPosition(position int) (Variant, error) {
+	for _, variant := range v {
+		if variant.Position == position {
+			return variant, nil
+		}
+	}
+
+	return Variant{}, NewErrVariantNotFoundByPosition()
+}
+
+// GetOption1Values returns all the option1 values for each variant if they have one
+func (v Variants) GetOption1Values() []string {
```

Comment:
What is this returning? Should it not be `ProductOption`? If not do you have an example of what is on variants and what is returned?

Reply:
This is a helper method that basically goes through all the variants and gives me the value of the option1 entries that can be used like this.

```golang
productOptions = append(productOptions, shopify.ProductOption{
		Name:   "Title",
		Values: shadowProduct.Variants.GetOption1Values(),
	})
```
This is because I want to do what the API wants me to do like this.

```json
product.options = [
  {
    "name": "Color",
    "values": [
      "Blue",
      "Black"
    ]
  },
  {
    "name": "Size",
    "values": [
      "155",
      "159"
    ]
  }
];
```